tests/test_inference_endpoints.py

```diff
@@ -244,6 +276,24 @@ def test_wait_failed(mock_get: Mock):
         endpoint.wait(refresh_every=0.001)
 
 
+@patch("huggingface_hub.hf_api.HfApi.get_inference_endpoint")
+def test_wait_update(mock_get: Mock):
+    """Test Waits Until Timeout Error Is Raised"""
+    endpoint = InferenceEndpoint.from_raw(MOCK_INITIALIZING, namespace="foo")
+
+    mock_get.side_effect = [
+        InferenceEndpoint.from_raw(MOCK_UPDATE, namespace="foo"),
+        InferenceEndpoint.from_raw(MOCK_UPDATE, namespace="foo"),
+        InferenceEndpoint.from_raw(MOCK_UPDATE, namespace="foo"),
+        InferenceEndpoint.from_raw(MOCK_UPDATE, namespace="foo"),
+    ]
+    with pytest.raises(InferenceEndpointTimeoutError):
+        endpoint.wait(timeout=0.1, refresh_every=0.05)
+
+    assert endpoint.status == "updating"
+    assert len(mock_get.call_args_list) >= 2
```

Comment:
This test is not wrong but I would rather test that if we have "update" ,"update, "update, "running" then the endpoint will indeed wait until the endpoint is running before returning. Prior to this PR, the `.wait` call would have returned immediately because `MOCK_UPDATE` has a `url` field. 

```suggestion
        InferenceEndpoint.from_raw(MOCK_RUNNING, namespace="foo"),
    ]
    endpoint.wait(refresh_every=0.05)
    assert endpoint.status == "running"
```

Reply:
Thanks For Suggestions. I Think There Are Not Sufficient Values For Effect As It Gives callback Error Every Time.
![Screenshot 2025-02-19 165715](https://github.com/user-attachments/assets/d284a211-a37e-439e-9abc-4511c9fd1c71)
Also We Need To Fake Status Code as 200.